plugins/source/aws/client/client.go

```diff
@@ -64,53 +64,33 @@ var errUnknownRegion = func(region string) error {
 }
 var errRetrievingCredentials = errors.New("error retrieving AWS credentials (see logs for details). Please verify your credentials and try again")
 
-func (s *ServicesManager) ServicesByPartitionAccountAndRegion(partition, accountId, region string) *Services {
-	if region == "" {
-		region = defaultRegion
-	}
-	return s.services[partition][accountId][region]
-}
-
-func (s *ServicesManager) ServicesByAccountForWAFScope(partition, accountId string) *Services {
-	return s.wafScopeServices[partition][accountId]
+func (s *ServicesManager) ServicesByPartitionAccount(partition, accountId string) *Services {
+	return s.services[partition][accountId]
 }
 
 func (s *ServicesManager) InitServices(details svcsDetail) {
-	if details.region != "" {
-		s.InitServicesForPartitionAccountAndRegion(details.partition, details.accountId, details.region, details.svcs)
-	} else {
-		s.InitServicesForPartitionAccountAndScope(details.partition, details.accountId, details.svcs)
-	}
+	s.InitServicesForPartitionAccount(details.partition, details.accountId, details.svcs)
 }
 
-func (s *ServicesManager) InitServicesForPartitionAccountAndRegion(partition, accountId, region string, svcs Services) {
+func (s *ServicesManager) InitServicesForPartitionAccount(partition, accountId string, svcs Services) {
 	if s.services == nil {
-		s.services = make(map[string]map[string]map[string]*Services)
+		s.services = make(map[string]map[string]*Services)
 	}
 	if s.services[partition] == nil {
-		s.services[partition] = make(map[string]map[string]*Services)
+		s.services[partition] = make(map[string]*Services)
 	}
 	if s.services[partition][accountId] == nil {
-		s.services[partition][accountId] = make(map[string]*Services)
+		s.services[partition][accountId] = &svcs
 	}
-	s.services[partition][accountId][region] = &svcs
-}
 
-func (s *ServicesManager) InitServicesForPartitionAccountAndScope(partition, accountId string, svcs Services) {
-	if s.wafScopeServices == nil {
-		s.wafScopeServices = make(map[string]map[string]*Services)
-	}
-	if s.wafScopeServices[partition] == nil {
-		s.wafScopeServices[partition] = make(map[string]*Services)
-	}
-	s.wafScopeServices[partition][accountId] = &svcs
+	s.services[partition][accountId].Regions = funk.IntersectString(s.services[partition][accountId].Regions, svcs.Regions)
```

Comment:
I might be missing something here, but I think this can be simplified to:

```
s.services[partition][accountId] = &svcs
```

(so no check for nil, and no `funk.IntersecString`). Otherwise, I don't understand how we can land in a situation where `InitServicesForPartitionAccount` is being called for a particular account, but has already been called for that same account before? And if we land in that situation somehow, is using the intersection of regions the right approach?

Reply:
Yeah, not considering `wafScopeServices`, just looking at the old definition of `InitServicesForPartitionAccountAndRegion`. Before, it was:

```
	if s.services[partition][accountId] == nil {
		s.services[partition][accountId] = make(map[string]*Services)
	}
	s.services[partition][accountId][region] = &svcs
```

But I think that nil check was to do map initialization, so we can then safely assign to the map afterwards. This function also got called once per region, so that makes sense.

However, now we have `InitServicesForPartitionAccount` which I believe should only get called once, and we don't need to do a nil check before assigning. Or am I missing something here?

Reply:
`InitServicesForPartitionAccount` Gets called for every single account in the list of Accounts specified by the user in the spec (or discovered via Orgs)... Accounts are only deduped by this map structure because the accountID/partitions are resolved in parallel in concurrent go routines.